_slides/deadlock_demolition.md

```diff
@@ -103,5 +98,17 @@ Chopsticks can be dirty or clean. Initially all chopsticks start out as dirty. F
 
 <horizontal />
 
+## Resource allocation graph
```

Comment:
I wonder if it makes more sense to have the RAG discussion before dining philosophers. Sorta feels like its more related to being a way to detect deadlock rather than being connected to dining philosophers. But I can see arguments for both ways..

Reply:
I think the original ordering is better, but added context is needed for "why" the RAG section is here (or just take it out entirely).

Personally I would lay it out as:

# Theory
## What is a deadlock?
## How can we detect deadlock?
### Introduce RAG
* Something about how a deadlock is 1-1 to a cycle in this graph

# Application
## Introduce Dining Philosophers  
### Model Dining Philosophers as a RAG
* Show how the "everyone grab left, then right" solution needs to a cycle in the RAG 
* Point out that since all deadlocks equate to a cycle in the RAG all solutions must stop said cycles from forming
### Go over solutions
* For each solution point out how it stops cycles from forming: 

ex. Mutex solution has all processes wait for a single mutex, which looks like a star graph -> no cycles.
semaphore of n-1 makes it so that it's impossible to form a complete cycle (one link is broken off)

Reply:
I agree that RAG should have more detail. Barring that, the current order has 2 problems:
1. 'Forward pointers' to definitions- we use deadlock and explain a 'use case' before we even define it, which makes little sense.
2. Interleaving of lab-relevant information (the RAG coverage) with lab-background (the Dining Philosophers introduction/answers).

An idea to meld the suggestions above would be to have a section where we show RAG's as an instance of Dining Philosophers-- I'd rather not give a simplified/shortened definition of the very general problem. So something like the following:

# Application

## RAG

### Definition / Usage

### Relation to Dining Philosophers

Reply:
To clarify, I'm pushing for the same order as in the PR, but with more slides on the RAG section.